pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -96,7 +97,28 @@ func (a *Actuator) Exists(ctx context.Context, machine *machinev1.Machine) (bool
 	// When create()/update() try to store machineSpec/status this might result in
 	// "Operation cannot be fulfilled; the object has been modified; please apply your changes to the latest version and try again."
 	// Therefore we don't close the scope here and we only store spec/status atomically either in create()/update()"
-	return newReconciler(scope).exists()
+	invalidMachineConfiguration, exists, err := newReconciler(scope).exists()
+	if !invalidMachineConfiguration {
+		return exists, err
+	}
+
+	if exists {
```

Comment:
Can this ever happen? Surely we don't know if the Machine exists when there is an invalid configuraiton?

Reply:
That is the situation with the machine being OK and we change to zone to some nonsense. The machine still exists, but we only want to Fail it.

Reply:
Ok, so this is a fake exists, we haven't actually checked with the cloud provider, we just think it should exist?

Perhaps it's better not to deal with that and just handle all `invalidMachineConfiguration` scenarios here

Reply:
I did it this way, because for the invalid machine, the phase is nil and we have to make sure it is failed. But if the machine was indeed valid before (exists) we want to fail it as well, but i don't want to assume what state is it in now.
IMO it is better to split the logic for used-to-be-valid/invalid machines.